src/code/z_effect_soft_sprite.c

```diff
@@ -27,17 +29,17 @@ void EffectSS_Clear(GlobalContext* globalCtx) {
     EffectSsOverlay* overlay;
     void* addr;
 
-    EffectSS2Info.data_table = NULL;
-    EffectSS2Info.searchIndex = 0;
-    EffectSS2Info.size = 0;
+    sEffectSsInfo.data_table = NULL;
+    sEffectSsInfo.searchIndex = 0;
+    sEffectSsInfo.size = 0;
 
-    // This code is completely useless, as data_table was just set to NULL and size to 0
-    for (effectsSs = EffectSS2Info.data_table; effectsSs < EffectSS2Info.data_table + EffectSS2Info.size; effectsSs++) {
+    //! @bug: This code is completely useless, as data_table was just set to NULL and size to 0
```

Comment:
Good job IDO.

Reply:
I don't think it's redundant. Because the table was just set to `NULL`, the individual effects don't get destroyed properly, i.e. `EffectSS_Delete` doesn't get called for each effect left in the table.

Reply:
Hmmm I see, then yeah that does kinda lean towards being a bug then. It'd be good to add that to the description of the bug to make clearer why it's a bug instead of useless/redundant

Reply:
Updated the comment to be clearer.